lib/build.js

```diff
@@ -1,73 +1,113 @@
-const exec = require('child_process').exec
-const fs = require('fs')
-const path = require('path')
-const purge = require('./purge')
+const exec = require("child_process").exec;
+const fs = require("fs");
+const path = require("path");
+const purge = require("./purge");
 
-// Conditionally require node-sass or dart-sass
-const pathToNodeSass = path.normalize(path.resolve(process.cwd(), 'node_modules', 'node-sass'))
-const pathToDartSass = path.normalize(path.resolve(process.cwd(), 'node_modules', 'sass'))
-let sass = null
+let sass = null;
+function getSass() {
+  // Conditionally require node-sass or dart-sass
+  const pathToNodeSass = path.normalize(
+    path.resolve(process.cwd(), "node_modules", "node-sass")
+  );
+  const pathToDartSass = path.normalize(
+    path.resolve(process.cwd(), "node_modules", "sass")
+  );
 
-try {
-  sass = require(pathToNodeSass)
-} catch (err) {
   try {
-    sass = require(pathToDartSass)
-  } catch (e) {
-    sass = null
+    sass = require(pathToNodeSass);
+  } catch (err) {
+    try {
+      sass = require(pathToDartSass);
+    } catch (e) {
+      sass = null;
+    }
   }
 }
 
-module.exports = ({ purgeFlag, configPath }) => {
-  const ngTwFile = configPath
-    ? path.normalize(path.resolve(process.cwd(), configPath))
-    : path.normalize(path.resolve(process.cwd(), 'ng-tailwind.js'))
-
-  const config = fs.existsSync(ngTwFile) && require(ngTwFile)
+function twBuild(sourceFile, config, configPath, purgeFlag) {
+  return new Promise((resolve, reject) =>
+    exec(
+      `${path.normalize(
+        "./node_modules/.bin/tailwind"
+      )} build "${sourceFile}" -c "${config.configJS}" -o "${
+        config.outputCSS
+      }"`,
+      err => {
+        if (err) return reject(err);
 
-  const twBuild = (sourceFile) => new Promise((resolve, reject) => exec(
-    `${path.normalize('./node_modules/.bin/tailwind')} build "${sourceFile}" -c "${config.configJS}" -o "${config.outputCSS}"`,
-    err => {
-      if (err) return reject(err)
+        console.info("Successful Tailwind Build!");
 
-      console.info('Successful Tailwind Build!')
+        if (config.purge || purgeFlag) purge({ configPath });
 
-      if (config.purge || purgeFlag) purge({ configPath })
+        return resolve(sourceFile);
+      }
+    )
+  );
+}
 
-      return resolve(sourceFile)
+function sassBuild(config, configPath) {
+  return new Promise((res, rej) => {
+    getSass();
```

Comment:
Why are we doing this on each call, instead of once when the script is loaded into memory?

Reply:
For sure, we can remove the `Promise` here in order to make it global as previous.